ts/private/ts_config.bzl

```diff
@@ -135,22 +127,30 @@ def _write_tsconfig_rule(ctx):
             extends_path = "./" + extends_path
         content = content.replace("__extends__", extends_path)
 
-    filtered_files = _filter_input_files(ctx.files.files, ctx.attr.allow_js, ctx.attr.resolve_json_module)
-
-    # Update file paths to be relative to the tsconfig file, including a ./ prefix
-    # to ensure paths are all relative to the config file.
-    package_prefix = ctx.label.package + "/"
-
-    # If the target is inside another workspace, we will need to also add the workspace
-    # root to the prefix.
+    # The prefix of source files that are within the same package as the tsconfig file.
+    local_package_prefix = "%s/" % ctx.label.package if ctx.label.package else ""
     if (len(ctx.label.repo_name) > 0):
-        package_prefix = "../{}/{}".format(ctx.label.repo_name, package_prefix)
-    filtered_files = [
-        "./" + f.short_path.removeprefix(package_prefix)
-        for f in filtered_files
-    ]
-
-    content = content.replace("\"__files__\"", str(filtered_files))
+        # If the target is inside another workspace the prefix also contains the navigation to that workspace.
+        local_package_prefix = "../{}/{}".format(ctx.label.repo_name, local_package_prefix)
+
+    # The path to navigate to the root of the workspace
+    path_to_root = "/".join([".." for _ in ctx.label.package.split("/")])
+
+    # Compute the list of source files with paths relative to the generated tsconfig file.
+    src_files = []
+    for f in ctx.files.files:
+        # Only include typescript source files
+        if not (_lib.is_ts_src(f.basename, ctx.attr.allow_js, ctx.attr.resolve_json_module) or _lib.is_typings_src(f.basename)):
+            continue
+
+        if f.short_path.startswith(local_package_prefix):
+            # Files within this project or subdirs can avoid the ugly ../ prefix
```

Comment:
does it actually matter? This output isn't for humans, it's fed directly to tsc, right? I'd maybe just go for the simpler code

Reply:
This PR doesn't change that so lets debate it another time